### PR TITLE
Feature/ms 05/mongoose/work profile

### PR DESCRIPTION
**Pull request by clementinob**
### Work Profile entity:
_Implemented Express Router, new Controller Methods, add folder "`model`"._

Request Methods added:
**GET** - "/"
**GET** - "/:id"
**GET** - "/byName/:name"
**POST** - "/create"
**PUT** - "/update/:id"
**DELETE** - "/delete/:id"

Controllers added:

- `getAllWorkProfiles()`
- `getWorkProfileByID()`
- `getWorkProfilesByName()`
- `createWorkProfile()`
- `updateWorkProfile()`
- `deleteWorkProfile()`

### CODE REVIEW

File: controllers/work-profiles.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      name: req.body.name,
+      description: req.body.description,
+    });
+    workProfile.save((error, workProfile) => {
+      if (error) {
+        return res.status(500).json({ msg: 'Error while creating a new Work Profile', error });
+      }
+      return res.status(201).json(workProfile);
+    });
+  };
```
Comment (aquilesmariani) [inline controllers/work-profiles.js:35]: There's no need to put an `else` in this part if you cover the conditional paths with the first `if` 
```suggestion
  if (!req.body.name || !req.body.description) {
    return res.status(400).send('Please include all fields to create a new Work Profile');
  }
  const workProfile = new WorkProfiles({
    name: req.body.name,
    description: req.body.description,
  });
  workProfile.save((error, workProfile) => {
    if (error) {
      return res.status(500).json({ msg: 'Error while creating a new Work Profile', error });
    }
    return res.status(201).json(workProfile);
  });
```

File: controllers/work-profiles.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    return res.status(400).json({ msg: `You should enter an ID to be deleted` });
+  } else {
+    WorkProfiles.findByIdAndDelete(req.params.id, (error, result) => {
+      if (error) {
+        return res.status(400).json(error);
+      } else {
+          if (result === null){
+            return res.status(404).json({ msg: 'ID not found' });
+          } else {
+            return res.status(200).json({ msg: `Work Profile with the id: ${req.params.id} was deleted!`, result });
```
Comment (aquilesmariani) [inline controllers/work-profiles.js:73]: Can we get rid of the `result` we are returning here? I don't see the point of sharing the data we just deleted.

File: controllers/work-profiles.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          }
+        }
+    });
+  }
+};
+
+module.exports = 
+  {
+    getAllWorkProfiles : getAllWorkProfiles,
```
Comment (aquilesmariani) [inline controllers/work-profiles.js:82]: Let's start using `property shorthands` to asign a property any value with the same name:
```suggestion
module.exports = {
    getAllWorkProfiles,
```

File: models/work-profiles.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
```
Comment (aquilesmariani) [inline models/work-profiles.js:2]: You can import it doing the following:
```suggestion
const { Schema } = mongoose;
```

File: models/work-profiles.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+const WorkProfileSchema = new Schema(
+		{
```
Comment (aquilesmariani) [inline models/work-profiles.js:5]: Please fix indentation here. You can also place every opening braces on the line above right after the parenthesis.

File: router/index.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
 const express = require('express');
-const { positionRouter } = require('./positions');
+const { workProfilesRouter } = require('./work-profiles.js');
 // const postulants = require('./postulants');
 const router = express.Router();
 
-router.use('/positions',positionRouter);
+router.use('/workProfiles',workProfilesRouter);
```
Comment (aquilesmariani) [inline router/index.js:6]: Separate params with a space after the comma
```suggestion
router.use('/workProfiles', workProfilesRouter);
```

**Review by aquilesmariani — CHANGES_REQUESTED** [review]

**Comment by aquilesmariani** [comment]
@clementinob you can also try running `npm run lint` to check the amount of errors you have in your files and fix them before requesting my review again

File: postman_collection.json [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+					},
+					"response": []
+				},
+				{
+					"name": "getWorkProfileByName",
+					"request": {
+						"method": "GET",
+						"header": [],
+						"url": {
+							"raw": "http://localhost:3000/api/workProfiles/6192da2beafe6937464852dc",
```
[the remark this answers is not in the document]
Comment (clementinob) [reply postman_collection.json:59]: Yup, didn't save the collection in Postman before the export. Done. Sorry about that.

File: controllers/work-profiles.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
+const WorkProfiles = require('../models/work-profiles.js');
+
+const getAllWorkProfiles = (req, res) => {
+  WorkProfiles.find()
+  .then((WorkProfiles) => res.status(200).json({WorkProfiles}))
+  .catch((error) => res.status(500).json(error));
```
[the remark this answers is not in the document]
Comment (clementinob) [reply controllers/work-profiles.js:6]: Done!

File: controllers/work-profiles.js (hunk 1 above)
Comment (clementinob) [reply controllers/work-profiles.js:35]: You are right. Done.

File: controllers/work-profiles.js (hunk 2 above)
Comment (clementinob) [reply controllers/work-profiles.js:73]: Now that you point it out to me, it makes sense to me. 👍

File: controllers/work-profiles.js (hunk 3 above)
Comment (clementinob) [reply controllers/work-profiles.js:82]: Right. Done!

File: models/work-profiles.js (hunk 4 above)
Comment (clementinob) [reply models/work-profiles.js:2]: I didn't know that it can be imported like this. Implemented ✅

File: models/work-profiles.js (hunk 5 above)
Comment (clementinob) [reply models/work-profiles.js:5]: ✔️

File: router/index.js (hunk 6 above)
Comment (clementinob) [reply router/index.js:6]: Done. Sorry about that. 👎

**Comment by clementinob** [comment]
@aquilesmariani Done with the Eslint!

**Review by aquilesmariani — APPROVED** [review]
Looks great!! Nice job!

**Pull request merged**